drivers/Aqara/aqara-presence-sensor/src/fp2/device_manager.lua

```diff
@@ -0,0 +1,272 @@
+local log = require "log"
+local json = require "st.json"
+local fields = require "fields"
+local capabilities = require "st.capabilities"
+local multipleZonePresence = require "multipleZonePresence"
+
+local PresenceSensor = capabilities.presenceSensor
+local MovementSensor = capabilities.movementSensor
+local ActivitySensor = capabilities.activitySensor
+local DeviceMode = capabilities["stse.deviceMode"]
+
+local MOVEMENT_TIMER = "movement_timer"
+local MOVEMENT_TIME = 5
+local COMP_MODE = "mode"
+
+local device_manager = {}
+device_manager.__index = device_manager
+
+local FP2_MODES = { "zoneDetection", "fallDetection", "sleepMonitoring" }
+
+function device_manager.presence_handler(driver, device, zone, evt_value)
+  if not device:supports_capability(PresenceSensor) then return end
+  local evt_action = "not present"
+  if evt_value == 1 then evt_action = "present" end
+  device:emit_event(PresenceSensor.presence(evt_action))
+end
+
+function device_manager.movement_handler(driver, device, zone, evt_value)
```

Comment:
In my testing I'm often not seeing this trigger and the attribute stays at "No movement", even as I walk around the room and my presence is detected as changing

Reply:
Could you confirm that it works the same in the Aqara Home app when you encounter that issue?
It's good to distinguish between issues with the status displayed in the SmartThings app and issues with FP2 devices behaving differently than expected.
If the SmartThings app is displaying the same status as the Aqara Home app, I would consider this an issue with the FP2 device itself.

Reply:
In the SmartThings app, I can only see the MovementSensor.movement events "approaching" and "movingAway", but not "entering", "leaving", "enteringLeft" and so on. The same as in the Aqara Home app logs: only approaching and moving away.

Reply:
If the logs are the same as the Aqara Home app, I would consider this an issue with the FP2 device rather than an issue with the edge driver.